lib/src/parsers/phone_parser.dart

```diff
@@ -42,6 +42,46 @@ class PhoneParser {
     return PhoneNumber(isoCode: isoCode, nsn: national);
   }
 
+  /// parses a [phoneNumber] given an [isoCode] when it is unknown whether the phone number is a local or an international one,
+  /// we just know it is specified in the country's locale
+  ///
+  /// {@macro phoneNumber}
+  ///
+  /// throws a PhoneNumberException if the isoCode is invalid
+  @internal
+  static PhoneNumber fromLocale(IsoCode isoCode, String phoneNumber) {

```

Comment:
I find the naming getting a bit confusing here, I'm not afraid to make breaking changes in order to establish a better terminology here, so if you have suggestions I'm all hears open.

Reply:
Yeah I also dislike adding yet another function as this makes things less intuitive to use. 
Personally I would recommend adapting the fromIsoCode to accept local and international numbers, I'm just not sure if that breaks anything and if that's okay with you.

Reply:
So I would propose that I adjust the behaviour of fromIsoCode such that it accepts all kinds of numbers those that are national to the given isoCode as well as international numbers (formatted according to isoCode as well as E164).

Reply:
What's your opinion on that? Should I add the functionality into the existing fromIsoCode, trying to keep the existing functionality?

Reply:
I don't think it should be merged but I could be wrong. currently it is like this:

`fromIsoCode(IsoCode.FR, ..)` will try to parse a phone number with the hint that it is supposed to be a phone number located in France.

Just a better terminology would fit better imo, the `fromLocale` name I think is confusing too since the first parameter in other functions is the location of the phone number. Here it's the location of the caller.

Reply:
Merged into fromRaw as proposed